### DISCUSSION TITLE
How can I test a websocket next ?

### DISCUSSION THREAD

**Discussion by cvgaviao** [quarkusio/quarkus#45287] [Q&A] [answered]
I'm creating a websocket for a use case where it seems that I just need to use the @OnOpen, to keep a stream being fulfilled at some ticket time, but I don't know to create a test for it.

There is nothing here related to websocket test here: https://quarkus.io/guides/websockets-next-tutorial

------

I've dug some test examples and found an interesting one [here](https://github.com/quarkusio/quarkus/blob/c3ccc1319270bf9f4659d858fce5343efcd82450/extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/pingpong/ServerPingClientPongTest.java).

But we can't use QuarkusUnitTest since its internal.
```
    @RegisterExtension
    public static final QuarkusUnitTest test = new QuarkusUnitTest()
            .withApplicationRoot(root -> {
                root.addClasses(EndpointText.class, EndpointBinary.class, WSClient.class);
            });
```

Could someone point me to some example ?

thanks

**Comment by cescoffier** [comment]
\CC @mkouba

**Reply by mkouba** [reply, marked as answer]
@cvgaviao You can use any WebSocket client to test your endpoint. In the [example linked](https://github.com/quarkusio/quarkus/blob/c3ccc1319270bf9f4659d858fce5343efcd82450/extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/pingpong/ServerPingClientPongTest.java) above we use the [Vert.x WebSocket client](https://vertx.io/docs/vertx-core/java/#_websockets_on_the_client) and you can use it too in your `@QuarkusTest` (there's no need to use the  `QuarkusUnitTest`). 

Another alternative is to use a [WebSocket Next client endpoint](https://quarkus.io/guides/websockets-next-reference#client-endpoints) located in the `src/test/java` so that it's not part of the production app; something along the lines of https://github.com/quarkusio/quarkus/blob/c3ccc1319270bf9f4659d858fce5343efcd82450/extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/client/ClientAutoPingIntervalTest.java.

There's also an [integration test](https://github.com/quarkusio/quarkus/blob/c3ccc1319270bf9f4659d858fce5343efcd82450/integration-tests/websockets-next/src/test/java/io/quarkus/websockets/ChatTest.java) for WS Next. And again, it's using the `WebSocketClient` from Vert.x.

**Reply by cvgaviao** [reply]
I'm going to try it. Thanks @mkouba !
